Sources/Internal/VisibleItemsProvider.swift

```diff
@@ -950,11 +950,13 @@ final class VisibleItemsProvider {
       daysAndFrames: dayRangeLayoutContext.daysAndFrames,
       boundingUnionRectOfDayFrames: dayRangeLayoutContext.boundingUnionRectOfDayFrames)
 
-    context.visibleItems.insert(
-      VisibleItem(
-        calendarItemModel: dayRangeItemProvider(dayRangeLayoutContext),
-        itemType: .dayRange(dayRange),
-        frame: frame))
+    if let dayRangeItemModel = dayRangeItemProvider(dayRangeLayoutContext) {
```

Comment:
`dayRangeItemProvider` and `overlaidItemLocationsAndItemProvider` are already optional on `CalendarViewContent`, so we can already gracefully handle them being nil.

Reply:
This creates a bit of an odd scenario. The way those 2 APIs work is that the first parameter is a set of date ranges / overlay locations that we're saying _will have_ a corresponding view. It really shouldn't be possible to return `nil`.

For example, if we do

```swift
CalendarViewContent(...)
  .dayRangeItemProvider(for: [selectedDateRange]) { ... }
```

then we really shouldn't be allowed to return `nil` from that closure. If we want to omit a day range, then we should omit it from that set of date ranges that we pass in as the first param.